src/devices/bus/centronics/epson_fx80.cpp

```diff
@@ -0,0 +1,337 @@
+{
+	epson_fx80_device::device_add_mconfig(config);
+	m_maincpu->set_addrmap(AS_PROGRAM, &epson_jx80_device::epson_jx80_mem);
+}
+
+
+//-------------------------------------------------
+//  INPUT_PORTS( epson_fx80 )
+//-------------------------------------------------
+
+INPUT_PORTS_START( epson_fx80 )
+INPUT_PORTS_END
+
+//-------------------------------------------------
+//  input_ports - device-specific input ports
+//-------------------------------------------------
+
+ioport_constructor epson_fx80_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME( epson_fx80 );
+}
+
+//-------------------------------------------------
+//  epson_fx80_device - constructor
+//-------------------------------------------------
+
+// constructor that passes device type
+epson_fx80_device::epson_fx80_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	epson_fx80_device(mconfig, EPSON_FX80, tag, owner, clock)
+{
+}
+
+// constructor to pass through the device type to device_t
+epson_fx80_device::epson_fx80_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_centronics_peripheral_interface(mconfig, *this),
+	m_maincpu(*this, "maincpu"),
+	m_slavecpu(*this, "slavecpu")
+{
+}
+
+// constructor that pass device type
+epson_jx80_device::epson_jx80_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: epson_fx80_device(mconfig, EPSON_JX80, tag, owner, clock)
+{ }
+
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void epson_fx80_device::device_start()
+{
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+
+void epson_fx80_device::device_reset()
+{
+}
+
+//-------------------------------------------------
+//  Slave CPU functions
+//-------------------------------------------------
+
+uint8_t epson_fx80_device::slave_r(offs_t offset)
+{
+	u8 const data = m_slavecpu->upi41_master_r(offset);
+	return data;
+}
+
+void epson_fx80_device::slave_w(offs_t offset, uint8_t data)
+{
+	if (offset == 0)
+		machine().scheduler().synchronize(
+			timer_expired_delegate(FUNC(epson_fx80_device::slave_write_data_sync), this), unsigned(data));
+	else if (offset == 1)
+		machine().scheduler().synchronize(
+			timer_expired_delegate(FUNC(epson_fx80_device::slave_write_command_sync), this), unsigned(data));
+}
+
+TIMER_CALLBACK_MEMBER(epson_fx80_device::slave_write_data_sync)
+{
+	m_slavecpu->upi41_master_w(0U, u8(u32(param)));
+}
+
+TIMER_CALLBACK_MEMBER(epson_fx80_device::slave_write_command_sync)
+{
+	m_slavecpu->upi41_master_w(1U, u8(u32(param)));
+}
+
+uint8_t epson_fx80_device::slave_t0_r()
+{
+	return home_sensor();
+}
```

Comment:
Why the trampoline?  Why not just attach the `home_sensor` placeholder function directly?

Reply:
ok, done.